crates/client/src/client/types.rs

```diff
@@ -169,3 +170,33 @@ impl TryFrom<OpaqueTransaction> for TransactionResponse {
         })
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum RelayedTransactionStatus {
+    Failed {
```

Comment:
Are there any other statuses besides failure we should keep track of?

Reply:
I think probably. But they can be defined in other Issues/PRs.